src/source/fromIterable.js

```diff
@@ -15,27 +15,16 @@ function IterableSource (iterable) {
 }
 
 IterableSource.prototype.run = function (sink, scheduler) {
-  return new IteratorProducer(getIterator(this.iterable), sink, scheduler)
+  return scheduler.asap(new PropagateTask(runProducer, getIterator(this.iterable), sink))
 }
 
-function IteratorProducer (iterator, sink, scheduler) {
-  this.scheduler = scheduler
-  this.iterator = iterator
-  this.task = new PropagateTask(runProducer, this, sink)
-  scheduler.asap(this.task)
-}
-
-IteratorProducer.prototype.dispose = function () {
-  return this.task.dispose()
-}
+function runProducer (t, iterator, sink) {
+  let r = iterator.next()
 
-function runProducer (t, producer, sink) {
-  var x = producer.iterator.next()
-  if (x.done) {
-    sink.end(t, x.value)
-  } else {
-    sink.event(t, x.value)
+  while (!r.done && this.active) {
```

Comment:
Yeah, this is slightly confusing.  When a PropagateTask executes the function you provided to its constructor, it [uses itself as the `thisArg`](https://github.com/cujojs/most/blob/master/src/scheduler/PropagateTask.js#L34).  It turned out to be useful to be able to look at the `active` state on several occasions, and for better or for worse (probably worse!), this was a way to "pass" the task itself without adding it to the argument list.  I could def see refactoring that at some point to pass something useful as an argument instead.